src/python/gem5/prebuilt/hifiveunmatched/hifive_proc.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2022 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from gem5.utils.override import overrides
+from gem5.components.boards.mem_mode import MemMode
+from gem5.components.processors.simple_core import SimpleCore
+
+from m5.util import warn
+
+from gem5.components.processors.abstract_processor import AbstractProcessor
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.isas import ISA
+from gem5.components.boards.abstract_board import AbstractBoard
+from hifive_core import U74Core
+
+from typing import Optional
+
+
+class U74Processor(AbstractProcessor):
+    """
+    A U74Processor contains a number of cores of U74Core.
+    """
+
+    def __init__(
+        self,
+    ) -> None:
+        self._cpu_type = CPUTypes.MINOR
+        super().__init__(
+            cores=self._create_cores()
+        )
+
+    def _create_cores(
+        self,
+    ):
+        return U74Core()
```

Comment:
Shouldn't this be `[U74Core() for i in range(4)]`?

Reply:
Hi Jason! Thanks for the feedback. I believe the MinorCPU in stdlib only runs one core. When we tried to set the number of cores as 4, it panicked. @jasrangi can you confirm that I recall correctly?

Reply:
You'll need to set them all up correctly 😆.

I guess 1 core is OK for now. But we need to be able to support the same configuration at some point.

Reply:
Oh I see, yeah I agree with you. We might need to set it up later. For now, even when we run with ./perf we taskset the workload so that there is no migration of CPU cores